block-layer/codecs/dag-pb.md

```diff
@@ -36,6 +36,21 @@ message PBNode {
 The objects link names are specified in the 'Name' field of the PBLink object.
 All link names in an object must either be omitted or unique within the object.
 
+This layout can be expressed with [IPLD Schemas](../../schemas/README.md) as:
+
+```ipldsch
+type PBLink struct {
+  hash optional Bytes
+  name optional String
+  tsize optional Int
+}
+
+type PBNode struct {
+  links [PBLink]
```

Comment:
I wonder `links [PBLink]` should be marked `optional`.  Does protobuf on the wire have distinguishment between an empty and an absent `repeated` element?

Reply:
> Does protobuf on the wire have distinguishment between an empty and an absent repeated element?

No. The "sub-frame" is either there or it isn't. There is no way to say on-wire "there are zero repeated elements of type N"

Reply:
Exactly:

* Required = exactly one (`.`).
* Optional = at most one (`.+`).
* Repeated = any number (`.*`).

Reply:
Therefore, there is no meaningful distinction between "empty" and "not there", so I wouldn't mark it as optional.